### PR TITLE
feat(trace): add TelemetryResourceDetector

### PR DESCRIPTION
**Pull request by TommyCpp**
https://github.com/open-telemetry/opentelemetry-demo/pull/504 needs to add `telemetry.sdk.*` info to the resources

### CODE REVIEW

File: opentelemetry-sdk/src/resource/telemetry.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+/// Note that the `telemetry.auto.version` is not provided as of now.
+///
+/// See [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md#telemetry-sdk) for details.
+#[derive(Debug)]
+pub struct TelemetryResourceDetector;
+
+impl ResourceDetector for TelemetryResourceDetector {
+    fn detect(&self, _timeout: Duration) -> Resource {
+        Resource::new(vec![
+            KeyValue::new("telemetry.sdk.name", "opentelemetry"),
```
Comment (hdost) [inline opentelemetry-sdk/src/resource/telemetry.rs:22]:
```suggestion
            KeyValue::new("telemetry.sdk.name", "opentelemetry-sdk"),
```

**Review by hdost — CHANGES_REQUESTED** [review]
Looks good! One minor comment.

**Review by hdost — APPROVED** [review]
The comment could honestly be done in a follow up.

File: opentelemetry-sdk/src/resource/telemetry.rs (hunk 1 above)
Comment (TommyCpp) [reply opentelemetry-sdk/src/resource/telemetry.rs:22]: Good call out, but the spec asks all opentelemetry provided SDK to use `opentelemetry` as the value for this key

> The default OpenTelemetry SDK provided by the OpenTelemetry project MUST set telemetry.sdk.name to the value opentelemetry

**Pull request merged**